backends/platform/libretro/src/libretro-os.cpp

```diff
@@ -0,0 +1,1329 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include <list>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <retro_inline.h>
+#include <retro_miscellaneous.h>
+
+#include "audio/mixer_intern.h"
+#include "backends/base-backend.h"
+#include "base/commandLine.h"
+#include "common/config-manager.h"
+#include "common/events.h"
+#include "common/tokenizer.h"
+#include "surface.libretro.h"
+
+#if defined(_WIN32)
+#include "backends/fs/windows/windows-fs-factory.h"
+#define FS_SYSTEM_FACTORY WindowsFilesystemFactory
+#else
+#include "libretro-fs-factory.h"
+#define FS_SYSTEM_FACTORY LibRetroFilesystemFactory
+#endif
+
+#include "backends/saves/default/default-saves.h"
+#include "backends/timer/default/default-timer.h"
+#include "graphics/colormasks.h"
+#include "graphics/palette.h"
+#if defined(_WIN32)
+#include <direct.h>
+#ifdef _XBOX
+#include <xtl.h>
+#else
+#include <windows.h>
+#endif
+#elif defined(__CELLOS_LV2__)
+#include <sys/sys_time.h>
+#elif (defined(GEKKO) && !defined(WIIU))
+#include <ogc/lwp_watchdog.h>
+#else
+#include <time.h>
+#endif
+
+#include "features/features_cpu.h"
```

Comment:
Please use the full path here and below.

Reply:
done